### PR TITLE
Enable strict parsing

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

This allows us to error when config has unknown fields.

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)

### CODE REVIEW

File: pkg/eks/api.go [hunk 1]
```diff
@@ -158,6 +159,10 @@ func LoadConfigFromFile(configFile string, cfg *api.ClusterConfig) error {
 		return errors.Wrapf(err, "reading config file %q", configFile)
 	}
 
+	if err := yaml.UnmarshalStrict(data, &api.ClusterConfig{}); err != nil {
```
Comment (martina-if) [inline pkg/eks/api.go]: Where is the old unmarshal call? was it the `runtime.Decode()` below? should we keep both?

Comment (errordeveloper) [reply pkg/eks/api.go]: Yes, and we still want to use that one. The output of this one is not stored. A bit of a hack, but couldn't find a way to make `runtime.Decode()` do the same, but it does other things that are useful for us.

Comment (martina-if) [reply pkg/eks/api.go]: Oh, but then this will be confusing, can we add a comment that the actual thing is used is `obj` from the `Decode()`?

Comment (errordeveloper) [reply pkg/eks/api.go]: Yes, good idea, I added a comment with more notes.

**Pull request merged**